bundles/org.platformio.eclipse.ide.home/src/org/platformio/eclipse/ide/home/core/BaseEnvironment.java

```diff
@@ -31,9 +31,11 @@
 import java.util.Map;
 import java.util.Optional;
 
+import org.osgi.service.component.annotations.Component;
 import org.platformio.eclipse.ide.home.api.CommandResult;
 import org.platformio.eclipse.ide.home.api.Environment;
 
+@Component
 public final class BaseEnvironment implements Environment {

```

Comment:
typically we use `Base` prefix for abstract classes and `Default` for the case like this